### PR TITLE
move min/max tests to TestTorchDeviceType

### PR DESCRIPTION
**Pull request by ngimel**
so that testing _min_max on the different devices is easier, and min/max operations have better CUDA test coverage.

### CODE REVIEW

File: test/test_torch.py [hunk 1]
```diff
@@ -15457,6 +15378,56 @@ def test_min_with_inf(self, device, dtype):
...
+                x[index] = nan
+                if not skip_indices:
+                    v, i = torchfn(x, 0)
+                    self.assertEqual(v, nan)
+                    self.assertEqual(i, index)
+                self.assertEqual(torchfn(x), nan)
+
+    @dtypesIfCPU(torch.float, torch.double, torch.long, torch.bool)
+    @dtypesIfCUDA(torch.half, torch.float, torch.long, torch.bool)
+    @dtypes(torch.float, torch.double)
```
Comment (mruberry) [inline test/test_torch.py:15415]: Technically these tests should still be decorated to be skipped if NumPy is not installed

File: test/test_torch.py [hunk 2]
```diff
@@ -15457,6 +15378,56 @@ def test_min_with_inf(self, device, dtype):
...
+    def test_max(self, device, dtype):
+        self._test_minmax_helper(torch.max, np.amax, device, dtype)
+
+    @dtypesIfCPU(torch.float, torch.double, torch.long, torch.bool)
+    @dtypesIfCUDA(torch.half, torch.float, torch.long, torch.bool)
+    @dtypes(torch.float, torch.double)
+    def test_min(self, device, dtype):
+        self._test_minmax_helper(torch.min, np.amin, device, dtype)
+
+    @onlyCPU
```
Comment (mruberry) [inline test/test_torch.py:15425]: Interesting - why do you want to run this variant on the CPU only?

File: test/test_torch.py [hunk 3]
```diff
@@ -15457,6 +15378,56 @@ def test_min_with_inf(self, device, dtype):
...
+                high = 2 if dtype == torch.bool else 1000
+                return torch.randint(low, high, shape, device=device, dtype=dtype)
+        x = create_input((100, 100), device, dtype)
+        self.compare_with_numpy(torchfn, reffn, x)
+        # non contiguous
+        x = create_input((10, 10, 10), device, dtype)
+        x = x[:, 4]
+        self.compare_with_numpy(torchfn, reffn, x)
+        # indices
+        if not skip_indices:
```
Comment (mruberry) [inline test/test_torch.py:15396]: You could create a tensor and its transpose as inputs, then enumerate over the product of dim x input here for more thorough coverage of various dims and the dim x noncontiguous case.

Comment (mruberry) [reply test/test_torch.py:15396]: (If you really wanted you to could product in keepdims, too

Comment (ngimel) [reply test/test_torch.py:15396]: Pass on keepdims, done for transpose and dim.

File: test/test_torch.py (hunk 2 above)
Comment (ngimel) [reply test/test_torch.py:15433]: Because _min_max is not yet implemented for cuda

File: test/test_torch.py (hunk 1 above)
Comment (ngimel) [reply test/test_torch.py:15421]: Done

File: test/test_torch.py (hunk 2 above)
Comment (mruberry) [reply test/test_torch.py:15433]: How about running it only on CPU and CUDA and asserting you get a runtime error on CUDA? That way if someone implements the CUDA behavior they'll know to update the test.

**Review by mruberry — APPROVED** [review]
Cool! I left a comment that you may want to assert _min_max fails on CUDA, too, instead of just skipping it.

**Pull request closed without merging**